### PR TITLE
[base] Added mapbox/io

### PR DESCRIPTION
**Pull request by brunoabinader**
Implemented `mapbox/io` as a header-only library that provides simple IO file operations: read, write, copy and delete are available at the moment.

### CODE REVIEW

File: mapbox/io/include/mapbox/io.hpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+namespace util {
+
+using ErrorType = std::string;
+
+nonstd::expected<std::string, ErrorType> readFile(const ghc::filesystem::path& filename) {
+    nonstd::expected<std::string, ErrorType> result;
+
+    std::ifstream file(filename.string(), std::ios::binary);
+    if (!file.good()) {
+        return nonstd::make_unexpected(std::string("Failed to open file ") + filename.string());
```
Comment (anderco) [inline mapbox/io/include/mapbox/io.hpp]: Maybe wrap filename in error message in single quotes, i.e.,

    Failed to open file 'foo.txt'

File: extras/CMakeLists.txt [hunk 2]
```diff
@@ -21,6 +16,11 @@ endfunction()
 add_library(mapbox-base-extras INTERFACE)
 add_library(Mapbox::Base::Extras ALIAS mapbox-base-extras)
 
+execute_process(
```
Comment (brunoabinader) [inline extras/CMakeLists.txt:19]: This is to avoid mapbox/io from attempting to fetch its non-existent git submodule.

**Review by anderco — COMMENTED** [review]
Will the source code for this library "live" in mapbox-base? I suppose it is so small it doesn't justify a repo of its own, but maybe we could have at least a README saying what it is about.

File: mapbox/io/include/mapbox/io.hpp (hunk 1 above)
Comment (brunoabinader) [reply mapbox/io/include/mapbox/io.hpp]: Done!

**Review by anderco — APPROVED** [review]
Great!

**Pull request merged**